VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/Attributes/OnPropertyChangedWithoutCallerMemberName/OnPropertyChangedWithoutCallerMemberNameCodeFix.cs

```diff
@@ -0,0 +1,97 @@
+        }
+
+        private async Task<Solution> AddCallerMemberNameAttribute(Document document, SyntaxNode root, SyntaxNode statement)
+        {
+            var editor = await DocumentEditor.CreateAsync(document);
+
+            var methodDeclaration = (MethodDeclarationSyntax)statement;
+            var param = methodDeclaration.ParameterList.Parameters.First();
+
+            var callerMemberNameAttribute = SyntaxFactory.Attribute(SyntaxFactory.ParseName("CallerMemberName"));
+            var attributeList = SyntaxFactory.AttributeList().AddAttributes(callerMemberNameAttribute);
+
+            var newParam = param.Default == null
+                ? param.WithAttributeLists(param.AttributeLists.Add(attributeList))
+                    .WithDefault(SyntaxFactory.EqualsValueClause(SyntaxFactory.ParseExpression("\"\"")))
+                : param.WithAttributeLists(param.AttributeLists.Add(attributeList));
+
+            editor.ReplaceNode(param, newParam);
+
+            var methodInvocations =
+                GetContainingClass(methodDeclaration).DescendantNodes()
+                    .OfType<InvocationExpressionSyntax>().Where(i =>
+                    {
+                        var identifierExpression = i.Expression as IdentifierNameSyntax;
+                        return identifierExpression != null && identifierExpression.Identifier.ValueText == "OnPropertyChanged";
+                    });
+
+            foreach (var methodInvocation in methodInvocations)
+            {
+                editor.ReplaceNode(methodInvocation,
+                    methodInvocation.WithArgumentList(SyntaxFactory.ArgumentList()));
+            }
+
+            var newRoot = await editor.GetChangedDocument().GetSyntaxRootAsync().ConfigureAwait(false);
+
+            var compilationUnit = (CompilationUnitSyntax)newRoot;
+
+            var usingSystemRuntimeCompilerServicesDirective = SyntaxFactory.UsingDirective(SyntaxFactory.ParseName("System.Runtime.CompilerServices"));
+            var usingDirectives = compilationUnit.Usings.Select(u => u.Name.GetText().ToString());
+
+            if (usingDirectives.All(u => u != usingSystemRuntimeCompilerServicesDirective.Name.GetText().ToString()))
+            {
+                var usings = compilationUnit.Usings.Add(usingSystemRuntimeCompilerServicesDirective).OrderBy(u => u.Name.GetText().ToString());
+
+                newRoot = newRoot.ReplaceNode(compilationUnit, compilationUnit.WithUsings(SyntaxFactory.List(usings))
+                    .WithAdditionalAnnotations(Formatter.Annotation));
+            }
+
+            var newDocument = document.WithSyntaxRoot(newRoot);
+            return newDocument.Project.Solution;
+        }
+
+        private ClassDeclarationSyntax GetContainingClass(MethodDeclarationSyntax method)
```

Comment:
You can just do `method.Ancestors().OfType<ClassDeclarationSyntax>().FirstOrDefault()`